gates.py

```diff
@@ -67,3 +73,9 @@ def __str__(self):
                        [0.0, 1.0, 0.0, 0.0],
                        [0.0, 0.0, 0.0, 1.0],
                        [0.0, 0.0, 1.0, 0.0]], dtype=np.complex256))
+
+SWAP = Gate('SWAP',
+            np.matrix([[1.0, 0.0, 0.0, 0.0],
+                       [0.0, 0.0, 1.0, 0.0],
+                       [0.0, 1.0, 0.0, 0.0],
+                       [0.0, 0.0, 0.0, 1.0]], dtype=np.complex256))
```

Comment:
Seems correct to me but it's worthy of short comment why this matrix is true. Something like 
```
# (a b) tensor (c d) = (ac bc ad bd). By swapping the middle two elements, producing 
# (ac ad bc bd) we can se by inspection that this is the result of (c d) tensor (a b).
# Hence we arrive at the matrix below that swap the middle elements of the 4-vector.
```
Or is there a more obvious way to realize the form of the swap matrix?